library/x509_create.c

```diff
@@ -37,44 +37,46 @@ typedef struct {
+    { ADD_STRLEN( "countryName" ),              MBEDTLS_OID_AT_COUNTRY,         MBEDTLS_ASN1_PRINTABLE_STRING },
+    { ADD_STRLEN( "O" ),                        MBEDTLS_OID_AT_ORGANIZATION,    MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "organizationName" ),         MBEDTLS_OID_AT_ORGANIZATION,    MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "L" ),                        MBEDTLS_OID_AT_LOCALITY,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "locality" ),                 MBEDTLS_OID_AT_LOCALITY,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "R" ),                        MBEDTLS_OID_PKCS9_EMAIL,        MBEDTLS_ASN1_IA5_STRING},
+    { ADD_STRLEN( "OU" ),                       MBEDTLS_OID_AT_ORG_UNIT,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "organizationalUnitName" ),   MBEDTLS_OID_AT_ORG_UNIT,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "ST" ),                       MBEDTLS_OID_AT_STATE,           MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "stateOrProvinceName" ),      MBEDTLS_OID_AT_STATE,           MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "emailAddress" ),             MBEDTLS_OID_PKCS9_EMAIL,        MBEDTLS_ASN1_IA5_STRING },
+    { ADD_STRLEN( "serialNumber" ),             MBEDTLS_OID_AT_SERIAL_NUMBER,   MBEDTLS_ASN1_PRINTABLE_STRING },
+    { ADD_STRLEN( "postalAddress" ),            MBEDTLS_OID_AT_POSTAL_ADDRESS,  MBEDTLS_ASN1_PRINTABLE_STRING },
+    { ADD_STRLEN( "postalCode" ),               MBEDTLS_OID_AT_POSTAL_CODE,     MBEDTLS_ASN1_PRINTABLE_STRING },
+    { ADD_STRLEN( "dnQualifier" ),              MBEDTLS_OID_AT_DN_QUALIFIER,    MBEDTLS_ASN1_PRINTABLE_STRING },
+    { ADD_STRLEN( "title" ),                    MBEDTLS_OID_AT_TITLE,           MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "surName" ),                  MBEDTLS_OID_AT_SUR_NAME,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "SN" ),                       MBEDTLS_OID_AT_SUR_NAME,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "givenName" ),                MBEDTLS_OID_AT_GIVEN_NAME,      MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "GN" ),                       MBEDTLS_OID_AT_GIVEN_NAME,      MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "initials" ),                 MBEDTLS_OID_AT_INITIALS,        MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "pseudonym" ),                MBEDTLS_OID_AT_PSEUDONYM,       MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "generationQualifier" ),      MBEDTLS_OID_AT_GENERATION_QUALIFIER, MBEDTLS_ASN1_UTF8_STRING },
+    { ADD_STRLEN( "domainComponent" ),          MBEDTLS_OID_DOMAIN_COMPONENT,   MBEDTLS_ASN1_IA5_STRING },
+    { ADD_STRLEN( "DC" ),                       MBEDTLS_OID_DOMAIN_COMPONENT,   MBEDTLS_ASN1_IA5_STRING },
+    { NULL, 0, NULL, MBEDTLS_ASN1_NULL }
 };
 
-static const char *x509_at_oid_from_name( const char *name, size_t name_len )
+static const x509_attr_descriptor_t *x509_at_oid_from_name( const char *name, size_t name_len )
```

Comment:
The name of this function is no longer accurate when it returns the entire descriptor, and not only the OID.

Reply:
What about `x509_at_descr_from_name`?